### PR TITLE
BugFix: Map Employee/Vendor before creating

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/netsuite/tasks.py [hunk 1]
```diff
@@ -93,12 +93,37 @@ def create_or_update_employee_mapping(expense_group: ExpenseGroup, netsuite_conn
...
+
+                if employee is None:
+                    created_entity: DestinationAttribute = netsuite_connection.post_employee(
+                        source_employee, auto_map_employees_preference, expense_group)
+
             else:
-                created_entity: DestinationAttribute = netsuite_connection.post_vendor(
-                    source_employee, auto_map_employees_preference, expense_group)
+                vendor = DestinationAttribute.objects.filter(
+                    attribute_type='VENDOR',
```
Comment (Shwetabhk) [inline apps/netsuite/tasks.py:119]: why not add this to filters itself, since it is condtional?

File: apps/netsuite/tasks.py [hunk 2]
```diff
@@ -93,12 +93,37 @@ def create_or_update_employee_mapping(expense_group: ExpenseGroup, netsuite_conn
...
+            elif auto_map_employees_preference == 'NAME':
+                filters = {
+                    'value__iexact': source_employee.detail['full_name']
+                }
+
             if employee_mapping_setting == 'EMPLOYEE':
-                created_entity: DestinationAttribute = netsuite_connection.post_employee(
-                    source_employee, auto_map_employees_preference, expense_group)
+                employee = DestinationAttribute.objects.filter(
+                    attribute_type='EMPLOYEE',
```
Comment (ashwin1111) [inline apps/netsuite/tasks.py:108]: sir, attribute type should be employee / vendor

Comment (Shwetabhk) [inline apps/netsuite/tasks.py:108]: why not add this to filters itself, since it is conditional?

Comment (ashwin1111) [reply apps/netsuite/tasks.py:108]: ignore

Comment (Shwetabhk) [reply apps/netsuite/tasks.py:108]: That way you'll have to write the query code only once

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**